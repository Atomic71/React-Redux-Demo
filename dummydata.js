export const dummyData = [
    {
        "id": 3,
        "title": "a new fresh todo list",
        "entries": [
            {
                "id": 8,
                "completed": false,
                "description": "a todo item 2",
                "todo_list": 3
            }, {
                "id": 7,
                "completed": true,
                "description": "a todo item",
                "todo_list": 3
            }
        ]
    }, {
        "id": 4,
        "title": "an even fresher todo list",
        "entries": [
            {
                "id": 1,
                "completed": true,
                "description": "Test item 1",
                "todo_list": 4
            }, {
                "id": 2,
                "completed": false,
                "description": "Test item 2",
                "todo_list": 4
            }, {
                "id": 3,
                "completed": false,
                "description": "some todo list item",
                "todo_list": 4
            }
        ]
    }
]
export const dummyDataWithoutListFour = [
    {
        "id": 3,
        "title": "a new fresh todo list",
        "entries": [
            {
                "id": 8,
                "completed": false,
                "description": "a todo item 2",
                "todo_list": 3
            }, {
                "id": 7,
                "completed": true,
                "description": "a todo item",
                "todo_list": 3
            }
        ]
    }
]

export const dummyDataWithNewList = [
    {
        "id": 3,
        "title": "a new fresh todo list",
        "entries": [
            {
                "id": 8,
                "completed": false,
                "description": "a todo item 2",
                "todo_list": 3
            }, {
                "id": 7,
                "completed": true,
                "description": "a todo item",
                "todo_list": 3
            }
        ]
    }, {
        "id": 4,
        "title": "an even fresher todo list",
        "entries": [
            {
                "id": 1,
                "completed": true,
                "description": "Test item 1",
                "todo_list": 4
            }, {
                "id": 2,
                "completed": false,
                "description": "Test item 2",
                "todo_list": 4
            }, {
                "id": 3,
                "completed": false,
                "description": "some todo list item",
                "todo_list": 4
            }
        ]
    }, {
        "id": 5,
        "title": "my new list",
        entries: []
    }
]

export const dummyArrOfObjects = [
    {
        "id": 8,
        "completed": false,
        "description": "a todo item 2",
        "todo_list": 3
    }, {
        "id": 7,
        "completed": true,
        "description": "a todo item",
        "todo_list": 3
    }, {
        "id": 1,
        "completed": true,
        "description": "Test item 1",
        "todo_list": 4
    }, {
        "id": 2,
        "completed": false,
        "description": "Test item 2",
        "todo_list": 4
    }, {
        "id": 3,
        "completed": false,
        "description": "some todo list item",
        "todo_list": 4
    }
];

export const dummyDataArrWithoutIdThree = [
    {
        "id": 8,
        "completed": false,
        "description": "a todo item 2",
        "todo_list": 3
    }, {
        "id": 7,
        "completed": true,
        "description": "a todo item",
        "todo_list": 3
    }, {
        "id": 1,
        "completed": true,
        "description": "Test item 1",
        "todo_list": 4
    }, {
        "id": 2,
        "completed": false,
        "description": "Test item 2",
        "todo_list": 4
    }
];

export const newItem = {
    "id": 88,
    "completed": false,
    "description": "some list item",
    "todo_list": 8
}

export const dummyArrOfObjectsWithNewItem = [
    {
        "id": 8,
        "completed": false,
        "description": "a todo item 2",
        "todo_list": 3
    }, {
        "id": 7,
        "completed": true,
        "description": "a todo item",
        "todo_list": 3
    }, {
        "id": 1,
        "completed": true,
        "description": "Test item 1",
        "todo_list": 4
    }, {
        "id": 2,
        "completed": false,
        "description": "Test item 2",
        "todo_list": 4
    }, {
        "id": 3,
        "completed": false,
        "description": "some todo list item",
        "todo_list": 4
    }, {
        "id": 88,
        "completed": false,
        "description": "some list item",
        "todo_list": 8
    }
];

export const dummyArrOfObjectsWithToggledThree = [
    {
        "id": 8,
        "completed": false,
        "description": "a todo item 2",
        "todo_list": 3
    }, {
        "id": 7,
        "completed": true,
        "description": "a todo item",
        "todo_list": 3
    }, {
        "id": 1,
        "completed": true,
        "description": "Test item 1",
        "todo_list": 4
    }, {
        "id": 2,
        "completed": false,
        "description": "Test item 2",
        "todo_list": 4
    }, {
        "id": 3,
        "completed": true,
        "description": "some todo list item",
        "todo_list": 4
    }
];

export const dummyArrOfObjectsWithUpdatedThree = [
    {
        "id": 8,
        "completed": false,
        "description": "a todo item 2",
        "todo_list": 3
    }, {
        "id": 7,
        "completed": true,
        "description": "a todo item",
        "todo_list": 3
    }, {
        "id": 1,
        "completed": true,
        "description": "Test item 1",
        "todo_list": 4
    }, {
        "id": 2,
        "completed": false,
        "description": "Test item 2",
        "todo_list": 4
    }, {
        "id": 3,
        "completed": false,
        "description": "updated list item",
        "todo_list": 4
    }
];