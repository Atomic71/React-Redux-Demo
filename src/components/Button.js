import React from 'react';
import PropTypes from 'prop-types'

export const Button = ({type, onClick, text}) => (
    <button className={`button ${type}`} onClick={onClick}>
        {text}
    </button>
)

Button.propTypes = {
    type: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    text: PropTypes.string.isRequired
}