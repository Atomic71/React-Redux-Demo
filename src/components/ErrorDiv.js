import React from 'react';
import {connect} from 'react-redux';
import {CSSTransition} from 'react-transition-group';
import PropTypes from 'prop-types';

import {hideErrorDiv} from '../store/actions/error/index';

const MapStateToProps = (state) => ({ error: state.errors.div });
const MapDispatchToProps = (dispatch) => ({ hideError: () => dispatch(hideErrorDiv()) })


export const ErrorDiv = ({ error, hideError }) => (
    <CSSTransition
        mountOnEnter
        unmountOnExit
        timeout={300} 
        in={error ? true : false} 
        classNames="slide" >
        <div key={error} className="error-div">
            <p className="error-text">
                {error}
            </p>
            <span onClick={hideError} className="error-close">&#10005;</span>
        </div>
    </CSSTransition>
)

ErrorDiv.propTypes = {
    error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]).isRequired,
    hideError: PropTypes.func.isRequired
}


export const ErrorDivConnected = connect(MapStateToProps, MapDispatchToProps)(ErrorDiv)