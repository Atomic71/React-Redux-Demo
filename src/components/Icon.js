import React from 'react';
import '../assets/icons.svg';

export const Icon = ({ icon, onClickHandler, itemTitle }) => (
    <svg className={`icon icon-${icon}`} onClick={onClickHandler}>
        <use xlinkHref={`#icons_icon-${icon}`}>
            <title>{itemTitle}</title>
        </use>
    </svg>
)