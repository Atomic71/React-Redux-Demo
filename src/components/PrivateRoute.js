import React, { Component } from 'react';
import { Route, Redirect, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

const MapStateToProps = (state) => ({
    token: state.userData.token
})

const PrivateRoute = ({ component: Component, token, ...rest }) => (
    <Route {...rest} render={(props) => (
        token
            ? <Component {...props} />
            : <Redirect to="/" />
    )}
    />
)

export default withRouter(connect(MapStateToProps)(PrivateRoute))