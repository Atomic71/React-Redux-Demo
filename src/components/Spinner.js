import React from 'react';

export const Spinner = () => (
    <div className="Spinner">
        <div className="bounce1"></div>
        <div className="bounce2"></div>
        <div className="bounce3"></div>
    </div>
)