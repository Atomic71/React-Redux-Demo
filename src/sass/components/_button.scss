.button {
    display: block;
    padding: .5rem 0;
    font-size: 1.3rem;
    outline: none;
    border: none;
    cursor: pointer;
    transition: all .3s ease-out;
    min-width: 200px;
    
    &-group {
        display: flex;
        left: 0;
        justify-content: center;
    }
    
    &.purple,
    &.purple-complementary:hover {
        background: rebeccapurple;
        color: whitesmoke;
    }
    &.primary {
        background: $dark;
        color: $white;
        &:hover {
            background: $dark-hover;
        }
    }
    &.processing {
        background: $grey;
        cursor: wait;
    }
    &.danger {
        background: $danger;
        &:hover {
            background: $danger-hover;
        }
    }
}
