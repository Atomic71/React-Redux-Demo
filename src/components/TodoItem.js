import React, {Component} from 'react';
import {Icon} from './Icon';
import { connect } from 'react-redux';

import { deleteItem, updateItem, toggleItem } from '../store/actions/item/index'
import autobind from 'autobind-decorator';
import { ChangeItemForm } from '../containers/forms/ChangeItem';
import { initialize } from 'redux-form'
 
const Item = ({item, toggleItem, changeItem, deleteItem}) => (
    <div 
        key={item.id} 
        className={`TodoItem ${item.completed ? "completed" : "" }`} >
        <p className="TodoItem-text">{item.description}</p>
        <div className="TodoItem-controls">
            <Icon onClickHandler={toggleItem} icon="clipboard" itemTitle="mark as completed" />
            <Icon onClickHandler={deleteItem} icon="bin" itemTitle="delete item" />
            <Icon onClickHandler={changeItem} icon="pencil" itemTitle="change name" />
        </div>
    </div>
)

const MapStateToProps = (state) => ({ token: state.userData.token })

const MapDispatchToProps = (dispatch) => {
    return {
        deleteItem: (item, token) => dispatch(deleteItem(item, token)),
        toggleItem: (item, token) => dispatch(toggleItem(item, token)),
        updateItem: (item, newName, token) => dispatch(updateItem(item, newName, token)),
        setInitialName: (currentName) => dispatch(initialize("ChangeItem", {NewName: currentName})) 
    }
}


class TodoItem extends Component {
    state = {
        isBeingEdited: false,
    }
    
    @autobind
    toggleViewHandler() { 
        this.setState(prevState => ( {isBeingEdited: !prevState.isBeingEdited} ))
    }

    @autobind
    prepareItemHandler() {
        const currentName = this.props.item.description;
        const { setInitialName } = this.props;
        setInitialName(currentName);
        this.toggleViewHandler();
    } 

    @autobind
    toggleItemHandler(){ 
        const { toggleItem, item, token} = this.props;
        toggleItem(item, token); 
    }
    
    @autobind
    deleteItemHandler(){ 
        const {deleteItem, item, token} = this.props;
        deleteItem(item, token);
    }

    render() {
        const {isBeingEdited, newName} = this.state;
        const {item} = this.props;
        let todoItem = <Item 
            item={item} 
            deleteItem={this.deleteItemHandler} 
            changeItem={this.prepareItemHandler}
            toggleItem={this.toggleItemHandler} 
        />

        isBeingEdited && ( todoItem = <ChangeItemForm 
            toggleView={this.toggleViewHandler}
            item={item} 
            />
        )

        return todoItem
    }
}

export const ConnectedTodoItem = connect(MapStateToProps,MapDispatchToProps)(TodoItem);