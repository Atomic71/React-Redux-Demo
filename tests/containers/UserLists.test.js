import React from 'react';
import {UserLists} from '../../src/components/UserListsAnimated';
import {dummyData} from '../../dummydata';
import renderer from 'react-test-renderer';
import MockRouter from 'react-mock-router';


describe("<UserLists />", () => {
    const lists = [ { "id": 3, "title": "a new fresh todo list" }, { "id": 4, "title": "an even fresher todo list" }]

    
    it("renders correctly when the lists are not fetched yet", () => {
        let tree = renderer
            .create(<UserLists/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("renders correctly when the lists are fetched but there are none", () => {
        let tree = renderer
            .create(<UserLists fetched lists={[]}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("renders correctly when the lists are fetched and there are lists", () => {
        let tree = renderer
            .create(<MockRouter><UserLists fetched lists={lists}/></MockRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})