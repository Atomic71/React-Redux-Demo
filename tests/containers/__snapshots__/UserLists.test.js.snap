// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserLists /> renders correctly when the lists are fetched and there are lists 1`] = `
<ul
  className="animatedList"
>
  <div
    className="list-link-wrapper"
  >
    <a
      className="list-link"
      href={undefined}
      onClick={[Function]}
    >
      a new fresh todo list
    </a>
  </div>
  <div
    className="list-link-wrapper"
  >
    <a
      className="list-link"
      href={undefined}
      onClick={[Function]}
    >
      an even fresher todo list
    </a>
  </div>
</ul>
`;

exports[`<UserLists /> renders correctly when the lists are fetched but there are none 1`] = `
<p>
  You have no lists!
</p>
`;

exports[`<UserLists /> renders correctly when the lists are not fetched yet 1`] = `
<div
  className="spinner"
>
  <div
    className="bounce1"
  />
  <div
    className="bounce2"
  />
  <div
    className="bounce3"
  />
</div>
`;
